test/bms-monitor-test.js

```diff
@@ -0,0 +1,33 @@
+const {expect} = require('chai');
+const assert = require('assert');
+const {batteryIsOk, SetLanguage} = require('../bms-monitor')
+const {getLog} = require('../bms-log')
+const batteryTestCases = [
+    ["ger","30 C", "30", "0.5", true, ['Temperatur OK: Normal','Ladezustand OK: Normal','Ladestrom OK: Normal']],
+    ["ger","20 C","30","0.80",false, ['Temperatur OK: Normal','Ladezustand OK: Normal','Ladestrom Warnung: Hoher Durchbruch steht bevor']],
+    ["en","40 C","21","0.3",false, ['Temperature OK: Normal','State of Charge Warning : approaching low Breach','Charge Rate OK: Normal']],
+    ["en","100 C", "38", "0.2", false, ['Temperature Breach : too high', 'State of Charge OK: Normal','Charge Rate OK: Normal' ]],
+    ["en","290 K", "19", "0.2", false, ['Temperature OK: Normal','State of Charge Breach : too low','Charge Rate OK: Normal']]
+]
+function BatteryTest(lang,temperature, soc, chargeRate, expected, expectedMessages){
+    it(`should return ${expected} with message ${expectedMessages} for inputs ${temperature}, ${soc} and  ${chargeRate}`,()=>{
+        SetLanguage(lang)
+        assert.equal(batteryIsOk(temperature, soc, chargeRate), expected)
+        const LogMessages = getLog(3)
```

Comment:
A good approach to data-driven testing and reducing the amount of test code :+1: 
But I found it hard to make out what's special about `3` over here.
So every time we add a new parameter, we will need to change all the test cases? I would expect to add a new test for the extra parameter and re-test the `batteryIsOk`, which puts all those results together.
Think of how you can reduce your unit-testing overhead. Hint: Test the parts and then test the whole.

Reply:
I created 2 unit tests  one for `checkStatus` to check the status of each parameter  and  another for `convertToCommonUnit` to check the coversion of paramters to common unit